cli/internal/helm/testdata/Azure/constellation-services/charts/kms/templates/daemonset.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    component: kms
+    k8s-app: kms
+    kubernetes.io/cluster-service: "true"
+  name: kms
+  namespace: testNamespace
+spec:
+  selector:
+    matchLabels:
+      k8s-app: kms
+  template:
+    metadata:
+      labels:
+        k8s-app: kms
+    spec:
+      containers:
+        - name: kms
+          image: ghcr.io/edgelesssys/constellation/kmsserver:v2.1.0@sha256:f56f901bd805550ac8232b4f7fdc8091a03ab16de5deddb1edd22b607413f406
```

Comment:
Regarding this image reference and all the others in the testdata:
I still feel like this is a maintenance burden. For testing purposes it would be very beneficial to inject image version lookup tables. I would not like to block this PR though. Maybe we can track this separately? I can create a ticket.

Reply:
I changed the HelmLoader struct to hold image references. The unittests then write bogus values into those fields.
The `New(..)` function needs at least the k8sVersion, unless we are fine with holding a pointer to the whole versionConfigs map in the HelmLoader. Therefore I needed to move the helmLoader instatiation into `initialize(..)`. This slows down the package tests for the `cmd` pkg by 0.1 secs because it loads the actual charts from the embed.FS. I guess this is fine(?), but I wanted to put it out there. 

Happy to hear any alternative solutions :)

Reply:
I think what you did now looks good to me. Thanks! :+1: